org/mozilla/jss/util/jssutil.c

```diff
@@ -593,6 +580,32 @@ JSS_ByteArrayToSECItem(JNIEnv *env, jbyteArray byteArray)
     return item;
 }
 
+/************************************************************************
+** JSS_ToByteArray.
+**
+** Converts the given chararacter array to a Java byte array.
+**
+** Returns
+**  The new jbyteArray object or NULL on failure.
+*/
+jbyteArray JSS_ToByteArray(JNIEnv *env, const void *data, int length)
```

Comment:
Since the data param is supposed to be a Java byte array, it might make more sense to declare it as const jbyte * data, but it's just nitpicking, the code is fine as is.

Reply:
I think the argument I make is that we use `JSS_ToByteArray` to store a lot of non-`jbyte` contents from time to time. Making the input `void *` ensures we don't need to cast the input parameter. 

Some examples are given in the second commit (where we use `JSS_ToByteArray`), but this includes storing pointers for `*Proxy` classes.

(Java/JNI lacks a true way of storing pointers managed by C code on a Java object. The best you can do is store the pointer as a `byte[]` and coerce to-from pointers).

Reply:
Trying out `jbyte *` instead of `void *` shows that you get a bunch of warnings:

```c
/home/cipherboy/GitHub/c2/jss/org/mozilla/jss/pkcs11/PK11Cipher.c: In function ‘Java_org_mozilla_jss_pkcs11_PK11Cipher_updateContext’:
/home/cipherboy/GitHub/c2/jss/org/mozilla/jss/pkcs11/PK11Cipher.c:166:37: warning: pointer targets in passing argument 2 of ‘JSS_ToByteArray’ differ in signedness [-Wpointer-sign]
     outArray = JSS_ToByteArray(env, outbuf, outlen);
                                     ^~~~~~
```